packages/walletconnect-connector/src/WalletConnectConnector.ts

```diff
@@ -197,49 +221,128 @@ export class WalletConnectConnector extends PredicateConnector {
             this.createModal();
             break;
           case 'CONNECT_SUCCESS': {
-            resolve(true);
-            unsub();
+            requestValidations();
             break;
           }
           case 'MODAL_CLOSE':
           case 'CONNECT_ERROR': {
+            if (wagmiConfig) {
+              const account = getAccount(wagmiConfig);
+              if (account) {
+                requestValidations();
+                break;
+              }
+            }
             resolve(false);
             unsub();
             break;
           }
         }
       });
     });
+    return result;
+  }
+
+  private async getAccountValidations(
+    accounts: `0x${string}`[] | string[],
+  ): Promise<boolean[]> {
+    return Promise.all(
+      accounts.map(async (a) => {
+        const isValidated = await this.storage.getItem(
+          `SIGNATURE_VALIDATION_${a}`,
+        );
+        return isValidated === 'true';
+      }),
+    );
+  }
+
+  private async accountHasValidation(
+    account: `0x${string}` | string | undefined,
+  ) {
+    if (!account) return false;
+    const [hasValidate] = await this.getAccountValidations([account]);
+    return hasValidate;
+  }
+
+  async requestValidations() {
+    const wagmiConfig = this.getWagmiConfig();
+    if (!wagmiConfig) {
+      throw new Error('Wagmi config not found');
+    }
+    const account = getAccount(wagmiConfig);
+    const { addresses } = account;
+    for (const address of addresses || []) {
+      await this.requestValidation(address)
+        .then(() => {
+          this.handleConnect(account);
+        })
+        .catch((err) => {
+          this.disconnect();
+          throw err;
+        });
+    }
+  }
+
+  async requestValidation(address?: string) {
+    return new Promise(async (resolve, reject) => {
+      // Disconnect if user dosen't provide signature in time
+      const validationTimeout = setTimeout(() => {
+        reject(
+          new Error("User didn't provide signature in less than 1 minute"),
+        );
+      }, SINGATURE_VALIDATION_TIMEOUT);
+      const { ethProvider } = await this.getProviders();
+
+      if (!ethProvider) return;
+
+      this.signAndValidate(ethProvider, address)
+        .then(() => {
+          clearTimeout(validationTimeout);
+          this.storage.setItem(`SIGNATURE_VALIDATION_${address}`, 'true');
+          resolve(true);
+        })
+        .catch((err) => {
+          clearTimeout(validationTimeout);
+          reject(err);
+        });
+    });
   }
 
   public async disconnect(): Promise<boolean> {
     const wagmiConfig = this.getWagmiConfig();
     if (!wagmiConfig) throw new Error('Wagmi config not found');
 
-    const { connector, isConnected } = getAccount(wagmiConfig);
+    const { addresses, connector, isConnected } = getAccount(wagmiConfig);
     await disconnect(wagmiConfig, {
       connector,
     });
-
+    addresses?.map((a) => this.storage.removeItem(`SIGNATURE_VALIDATION_${a}`));
```

Comment:
if we always remove this on disconnect there shouldn't be a need to store more than one, since it only stores for the active account.